### PR TITLE
Optimize ptr

### PR DESCRIPTION
**Pull request by jacquesqiao**
- CreateOp should return std::shared_ptr<OperatorBase> for python API
- Use ScopePtr to shorten std::shared_ptr<Scope>

### CODE REVIEW

File: paddle/framework/net.h [hunk 1]
```diff
@@ -69,7 +69,7 @@ class Net {
    * environment for ops. `begin` and `end` specify the scope of `ops_` to run,
    * If no positive indexes are provided, all operators in `ops_` will run.
    */
-  virtual void Run(std::shared_ptr<Scope> scope, DeviceContext *ctx) = 0;
+  virtual void Run(ScopePtr scope, DeviceContext *ctx) = 0;
```
Comment (reyoung) [inline paddle/framework/net.h]: const ScopePtr&

File: paddle/framework/op_registry_test.cc [hunk 2]
```diff
@@ -88,8 +89,7 @@ TEST(OpRegistry, IllegalAttr) {
 
   bool caught = false;
   try {
-    paddle::framework::OperatorBase* op __attribute__((unused)) =
-        paddle::framework::OpRegistry::CreateOp(op_desc);
+    OpPtr op __attribute__((unused)) = OpRegistry::CreateOp(op_desc);
```
Comment (reyoung) [inline paddle/framework/op_registry_test.cc]: ??
GTEST has a `ASSERT_THROW` macro.

Maybe in following PRs that part could be changed.

File: paddle/framework/operator.h [hunk 3]
```diff
@@ -30,7 +30,7 @@ namespace paddle {
 namespace framework {
 
 class OperatorBase;
-
+using OpPtr = std::shared_ptr<OperatorBase>;
```
Comment (reyoung) [inline paddle/framework/operator.h]: Maybe `OperatorPtr` is a better name for consistency.

Comment (jacquesqiao) [inline paddle/framework/operator.h]: done

File: paddle/framework/net.h (hunk 1 above)
Comment (jacquesqiao) [inline paddle/framework/net.h]: done

**Pull request merged**